app/src/main/java/com/nexenio/bleindoorpositioningdemo/ui/beaconview/map/BeaconMap.java

```diff
@@ -122,39 +125,42 @@ protected void drawDevice(Canvas canvas) {
         drawDeviceHistory(canvas);
         drawDevicePrediction(canvas);
 
-        PointF deviceCenter = (deviceLocationAnimator == null) ? canvasCenter : getPointFromLocation(deviceLocationAnimator.getLocation());
+        if (deviceLocationAnimator == null) {
+            return;
+        }
+
+        PointF deviceCenter = getPointFromLocation(deviceLocationAnimator.getLocation());
 
-        float deviceAdvertisingRange = 20; // in meters TODO: get real value based on tx power
-        float advertisingRadius = (float) canvasProjection.getCanvasUnitsFromMeters(deviceAdvertisingRange);
+        float locationAccuracy = (float) deviceLocationAnimator.getLocation().getAccuracy();
+        float deviceAccuracyRadius = (float) canvasProjection.getCanvasUnitsFromMeters(locationAccuracy);
 
-        // TODO use this for accuracy visualization
-        float animationValue = (deviceAccuracyAnimator == null) ? 0 : (float) deviceAccuracyAnimator.getAnimatedValue();
+        float animationValue = (deviceRadiusAnimator == null) ? 0 : (float) deviceRadiusAnimator.getAnimatedValue();
         float strokeRadius = (pixelsPerDip * 10) + (pixelsPerDip * 2 * animationValue);
 
-        //canvas.drawCircle(deviceCenter.x, deviceCenter.y, strokeRadius, deviceRangePaint);
-        canvas.drawCircle(deviceCenter.x, deviceCenter.y, advertisingRadius, deviceRangePaint);
+        canvas.drawCircle(deviceCenter.x, deviceCenter.y, deviceAccuracyRadius, deviceRangePaint);
         canvas.drawCircle(deviceCenter.x, deviceCenter.y, strokeRadius, whiteFillPaint);
         canvas.drawCircle(deviceCenter.x, deviceCenter.y, strokeRadius, secondaryStrokePaint);
         canvas.drawCircle(deviceCenter.x, deviceCenter.y, pixelsPerDip * 8, secondaryFillPaint);
     }
 
     protected void drawDeviceHistory(Canvas canvas) {
-        float accuracy = 1; // in meters TODO: get real value
-        float advertisingRadius = (float) canvasProjection.getCanvasUnitsFromMeters(accuracy);
-        Paint historyFillPaint = new Paint(secondaryFillPaint);
+        PointF deviceCenter;
+        float heatmapRadius = (float) canvasProjection.getCanvasUnitsFromMeters(1);
```

Comment:
In the old implementation we wanted to use the `accuracy` instead of `1`. Why did that change?

Reply:
Because the heatmap looked strange with that (I tried it). We use the circle around the device to visualize the accuracy.